classes/Product.php

```diff
@@ -4090,7 +4090,7 @@ public static function isDiscounted($id_product, $quantity = 1, Context $context
 
         $id_currency = (int) $context->currency->id;
         $ids = Address::getCountryAndState((int) $context->cart->{Configuration::get('PS_TAX_ADDRESS_TYPE')});
-        $id_country = $ids['id_country'] ? (int) $ids['id_country'] : (int) Configuration::get('PS_COUNTRY_DEFAULT');
+        $id_country = ($ids && $ids['id_country']) ? (int) $ids['id_country'] : (int) Configuration::get('PS_COUNTRY_DEFAULT');
```

Comment:
```suggestion
        $id_country = (int) ($ids['id_country'] ?? Configuration::get('PS_COUNTRY_DEFAULT'));
```

Reply:
Applied with the null coalescing operator, lovely. Thanks :)